Cabal/src/Distribution/Simple/Compiler.hs

```diff
@@ -363,6 +364,14 @@ unitIdSupported = ghcSupported "Uses unit IDs"
 backpackSupported :: Compiler -> Bool
 backpackSupported = ghcSupported "Support Backpack"
 
+-- | Does this compiler support the -jsem option
+jsemSupported :: Compiler -> Bool
+jsemSupported comp = case compilerFlavor comp of
+  GHC -> v >= mkVersion [9, 7]
+  _ -> False
```

Comment:
Never a fan of catch-alls, they can bite us in the future (even though in this case it is low risk).

Reply:
With the consolidation of backend behinds GHC, I have no fears of that one specifically.